core/src/test/java/io/questdb/test/griffin/SqlOptimiserTest.java

```diff
@@ -149,6 +149,23 @@ public void testAliasAppearsInFuncArgs4() throws Exception {
         });
     }
 
+
+    @Test
+    public void testQueryPlanForLastAggregateFunctionOnDesignatedTimestampColumn() throws Exception {
+        assertMemoryLeak(() -> {
+            ddl("create table y ( x int, ts timestamp) timestamp(ts);");
+            final String query = "select LAST(ts), x from y";
+            final QueryModel model = compileModel(query);
+            TestUtils.assertEquals("select-choose ts, x from (select [ts, x] from y timestamp (ts)) order by ts desc limit 1", model.toString0());
```

Comment:
This doesn't look correct. The query is an implicit GROUP BY which returns all <x, last(ts)> groups.

Reply:
@puzpuzpuz as the scope for this issue is it required to do a backward scan per group or **if there is a select on only designated timestamp column** 
For eg-:
Select timestamp from table_name?

Reply:
The issue is only for queries like `select last(ts) from y` / `select first(ts) from y`, not for `select last(ts), x from y` / `select first(ts), x from y` or any other kind of keyed GROUP BY or GROUP BY involving multiple aggregate functions. For instance, `select LAST(ts), x from y` cannot be rewritten into `select ts, x from y limit -1` since that's not correct.